typescript/lib/networks/network_context.ts

```diff
@@ -0,0 +1,628 @@
+
+        const [serviceContext, hostPortBindings, err] = await this.addServiceToPartition(
+            serviceId,
+            defaultPartitionId,
+            containerCreationConfig,
+            generateRunConfigFunc,
+        );
+        if (err !== null) {
+            return [null, null, err];
+        }
+
+        return [serviceContext, hostPortBindings, null];
+    }
+
+    // Docs available at https://docs.kurtosistech.com/kurtosis-libs/lib-documentation
+    public async addServiceToPartition(
+        serviceId: ServiceID,
+        partitionId: PartitionID,
+        containerCreationConfig: ContainerCreationConfig,
+        generateRunConfigFunc: (ipAddr: string, generatedFileFilepaths: Map<string, string>, staticFileFilepaths: Map<StaticFileID, string>) => [ContainerRunConfig, Error],
+    ): Promise<[ServiceContext, Map<string, PortBinding>, Error]> {
+
+        winston.info("Registering new service ID with Kurtosis API..."); //TODO logrus.Trace is meant for something for low level, but I could find winston.info to be closest to this
+        const registerServiceArgs: RegisterServiceArgs = newGetRegisterServiceArgs(serviceId, partitionId);
+        
+        // TODO TODO TODO - CALLBACK & ERROR-HANDLING (Remove)
+        // registerServiceResp, err := networkCtx.client.RegisterService(ctx, registerServiceArgs)
+        // if err !== nil {
+        //     return nil, nil, stacktrace.Propagate(
+        //         err,
+        //         "An error occurred registering service with ID '%v' with the Kurtosis API",
+        //         serviceId)
+        // }
+
+        const promiseAsyncRegisterService: Promise<ResultAsync<RegisterServiceResponse, Error>> = new Promise((resolve, _unusedReject) => { //TODO
+            this.client.registerService(registerServiceArgs, (_unusedError: grpc.ServiceError, response: RegisterServiceResponse) => {
+                resolve(okAsync(response));
+            })
+        });
+        const promiseRegisterService: Result<RegisterServiceResponse, Error> = await promiseAsyncRegisterService;
+        if (!promiseRegisterService.isOk()) {
+            return [null, null, promiseRegisterService.error];
+        }
+        const registerServiceResp: RegisterServiceResponse = promiseRegisterService.value;
+
+        const serviceIpAddr = registerServiceResp.getIpAddr();
+
+        const serviceContext: ServiceContext = new ServiceContext(
+            this.client,
+            serviceId,
+            serviceIpAddr,
+            this.enclaveDataVolMountpoint,
+            containerCreationConfig.getKurtosisVolumeMountpoint());
+        winston.info("New service successfully registered with Kurtosis API");
+
+        winston.info("Loading static files into new service namespace...");
+        const usedStaticFilesMap = containerCreationConfig.getUsedStaticFiles();
+        //TODO new section to stick with a map, but this doens't feel productive though (4 lines below)
```

Comment:
Yep - you can swap this all for `usedStaticFilesMap.keys()` (this is very common convenience available in maps in all modern languages)

Reply:
Unfortunately, `usedStaticFilesMap.keys()` gives back a type of `IterableIterator<string>` while we need type `Set<string>`. I think we would need to manually create the Set either way.